internal/store/postgres/resource_repository.go

```diff
@@ -213,3 +214,52 @@ func (r ResourceRepository) GetByURN(ctx context.Context, urn string) (resource.
 
 	return resourceModel.transformToResource(), nil
 }
+
+func buildGetResourcesByNamespaceQuery(dialect goqu.DialectWrapper, withResourceType bool) (string, error) {
+	namespaceQueryExpression := goqu.Ex{
+		"backend": goqu.L("$2"),
+	}
+
+	if withResourceType {
+		namespaceQueryExpression["resource_type"] = goqu.L("$3")
+	}
+
+	getNamespaceQuery := dialect.Select("id").From(TABLE_NAMESPACES).Where(namespaceQueryExpression)
+	getResourcesByURNQuery, _, err := dialect.Select(&ResourceCols{}).From(TABLE_RESOURCES).Where(goqu.Ex{
+		"name":         goqu.L("$1"),
+		"namespace_id": goqu.Op{"in": getNamespaceQuery},
+	}).ToSQL()
+
+	return getResourcesByURNQuery, err
+}
+
+func (r ResourceRepository) GetByNamespace(ctx context.Context, name string, ns namespace.Namespace) (resource.Resource, error) {
+	var fetchedResource Resource
+
+	getResourceByNamespace, err := buildGetResourcesByNamespaceQuery(dialect, ns.ResourceType != "")
+	if err != nil {
+		return resource.Resource{}, fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	if ns.ResourceType == "" {
+		err = r.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+			return r.dbc.GetContext(ctx, &fetchedResource, getResourceByNamespace, name, ns.Backend)
+		})
+	} else {
+		err = r.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+			return r.dbc.GetContext(ctx, &fetchedResource, getResourceByNamespace, name, ns.Backend, ns.ResourceType)
+		})
+	}
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return resource.Resource{}, resource.ErrNotExist
+	} else if err != nil {
+		return resource.Resource{}, fmt.Errorf("%w: %s", dbErr, err)
+	}
+
+	if err != nil {
+		return resource.Resource{}, err
+	}
```

Comment:
Here, you can just use the result like this

```suggestion
	query, params, err := buildGetResourcesByNamespaceQuery(dialect, ns.ResourceType != "")
	if err != nil {
		return resource.Resource{}, fmt.Errorf("%w: %s", queryErr, err)
	}

	if err = r.dbc.WithTimeout(ctx, func(ctx context.Context) error {
		return r.dbc.GetContext(ctx, &fetchedResource, query, params...)
	}); err != nil {
	        switch {
	            case errors.Is(err, sql.ErrNoRows) {
		        return resource.Resource{}, resource.ErrNotExist
	            default:
	                 return resource.Resource{}, fmt.Errorf("%w: %s", dbErr, err)
	         }
	}
```

Reply:
resolved